src/api/app/models/backend_info.rb

```diff
@@ -9,30 +9,22 @@ class BackendInfo < ApplicationRecord
   #### Scopes (first the default_scope macro if is used)
   #### Validations macros
   #### Class methods using self. (public and then private)
-  def self.set_value(key, value)
-    backend_value = BackendInfo.find_or_initialize_by(key: key)
-    backend_value.value = value
-    backend_value.save!
-  end
-
-  def self.lastnotification_nr=(nr)
-    set_value('lastnotification_nr', nr.to_s)
-  end
-
-  def self.get_value(key)
-    BackendInfo.where(key: key).pluck(:value)
-  end
-
-  def self.get_integer(key)
-    nr = get_value(key)
-    nr.empty? ? 0 : nr[0].to_i
-  end
+  #### To define class methods as private use private_class_method
+  def self.key_accessors(*keys)
+    keys.each do |key|
+      define_singleton_method key do
+        BackendInfo.where(key: key).pluck(:value).first.to_i
+      end
 
-  def self.lastnotification_nr
-    get_integer('lastnotification_nr')
+      define_singleton_method "#{key}=" do |value|
+        backend_value = BackendInfo.find_or_initialize_by(key: key)
+        backend_value.value = value
+        backend_value.save!
+      end
```

Comment:
IMHO this makes the code harder to understand. What about just merging the methods so its like this?

```ruby
  def self.lastnotification_nr=(nr)
    backend_value = find_or_initialize_by(key: 'lastnotification_nr')
    backend_value.value = nr.to_s
    backend_value.save!
  end

  def self.lastnotification_nr
    nr = where(key: 'lastnotification_nr').pluck(:value)
    nr.empty? ? 0 : nr[0].to_i  end
  end
```

Reply:
Maybe harder to understand but more flexible! But I will merge it!